src/include/indexlr-workers.h

```diff
@@ -286,7 +286,7 @@ MinimizeWorker::work()
 				size_t sharpPos = read.id.find('#');
 				if (sharpPos != std::string::npos) {
 					size_t slashPos = read.id.rfind('/');
-					if (slashPos > sharpPos) {
+					if (slashPos != std::string::npos && slashPos > sharpPos) {
```

Comment:
It's okay if the read ID does not contain a slash. The `/1` and `/2` are optional. In that case, the barcode goes from the `#` to the end of the read ID.

Reply:
oh in that case, no explicit handling is required. `Number of characters to include in the substring (if the string is shorter, as many characters as possible are used).` frpm www.cplusplus.com/reference/string/string/substr/

Reply:
but if that is the case should I still use `rfind`? Isn't there a possibility the read identifier can contain "/". I don't think that is an illegal character.

Reply:
Anything's possible, but the read identifier does not usually contain `/` except for the `/1` and `/2` at the end. I'm okay with using `rfind`. You could also check whether the second-to-last character is a `/`.

Reply:
so if slash isn't necessary, there isn't a need to handle the case where `slashPos ==std::string::npos` because `string.substr` will retrieve as many characters as possible if the number of characters exceed length of string